src/authInfo.ts

```diff
@@ -921,12 +885,11 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
   // Build OAuth config for a JWT auth flow
   private async buildJwtConfig(options: OAuth2Options): Promise<AuthFields> {
     const privateKeyContents = await fs.readFile(ensure(options.privateKey), 'utf8');
-    const audienceUrl = getJwtAudienceUrl(options);
     const jwtToken = jwt.sign(
       {
         iss: options.clientId,
         sub: this.getUsername(),
-        aud: audienceUrl,
+        aud: options.loginUrl ?? process.env.SFDX_AUDIENCE_URL ?? SfdcUrl.PRODUCTION,
```

Comment:
without having to handle internal urls and looking at the created org instance, not sure we need this in other methods at all.  This hopefully makes clear the prioritization.

The original didn't seem to reference it for audiences, but I think we should include the loginUrl from the project file here, too: https://github.com/forcedotcom/schemas/blob/c0df4d670cd368d2cf760064983b8fa2ca04a4a8/sfdx-project.schema.json#L99

Reply:
I'm almost positive this order will not work for the majority of orgs.